src/parking-violations-reporting/use-parking-violations.tsx

```diff
@@ -1,24 +1,73 @@
-import {useUserLocation} from './use-user-location';
-import {useInitQuery} from './use-init-query';
+import {useCallback, useEffect, useState} from 'react';
+import {useGeolocationState} from '@atb/GeolocationContext';
+import {initViolationsReporting} from '@atb/api/mobility';
+import {
+  ParkingViolationType,
+  ViolationsReportingProvider,
+} from '@atb/api/types/mobility';
+import {Coordinates} from '@entur/sdk';
+
+export class PermissionRequiredError extends Error {
+  constructor() {
+    super('Permission required');
+  }
+}
 
 export const useParkingViolations = () => {
-  const {position, error: locationError} = useUserLocation();
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState<Error>();
+  const [violations, setViolations] = useState<ParkingViolationType[]>([]);
+  const [providers, setProviders] = useState<ViolationsReportingProvider[]>([]);
+  const [position, setPosition] = useState<Coordinates>();
+
   const {
-    isLoading,
-    error: loadingError,
-    data,
-  } = useInitQuery(position ?? {latitude: 0, longitude: 0});
+    getCurrentPosition,
+    status: locationPermissionStatus,
+    requestPermission,
+  } = useGeolocationState();
+
+  const requestLocationPermission = useCallback(requestPermission, []);
+
+  useEffect(() => {
+    if (locationPermissionStatus !== 'granted') {
```

Comment:
In this case, the `locationPermissionStatus` could be `.denied` not prompting anything to the user because it was already rejected. I think we could use a warning message indicating that location is not being shared. I assume that this prompt will just pop up without any communication, for the case when status is `notDetermined`. 

NOTE: There is a PR that will prompt an alert indicating go to the settings.

Reply:
I'm not sure if I follow. What this code currently does is request permission again if the `locationPermissionStatus` is any other than `granted`. If that request is not granted the warning message indicating that the location is not shared is shown.  (`PermissionRequiredError` is picked up in `Root_ParkingViolationsSelect.tsx`)